_posts/zh/newsletters/2019-05-29-newsletter.md

```diff
@@ -0,0 +1,179 @@
+
+    本周，John Newbery [建议][smaller v1 spk]我们使用某种方法避免在 scriptPubKey 中放置此字节。Wuille 同意这可能有用，并将尝试实现一种变体，其中该位将作为 taproot 见证数据的一部分。这将减少创建一个 taproot 输出的成本一个 vbyte（使其与当前的 P2WSH 相同）。
+
+- **<!--presentation-a-return-to-fees-->****演讲：重返费用：** 本月早些时候在纽约区块链周期间，Bitcoin Optech 贡献者 Mike Schmidt 在 Optech 的首次高管简报会上做了关于比特币交易费用的演讲。[他演讲的视频][vid return to fees]现已上线。{{return-to-fees}}
+
+## 提议的交易输出承诺
+
+提议的 [opcode `OP_CHECKOUTPUTSHASHVERIFY`][bip-coshv] 允许 Taproot 地址承诺一个或多个 tapscripts，这些 tapscripts 要求花费它们的交易包含一组特定的输出，这种技术在合同协议研究者中称为*契约*。该提议的操作码的主要好处是允许在当前费用较高时确认一笔小交易，并信任地保证一组人在未来费用较低时收到实际付款。对于已经实施了诸如[支付批处理][topic payment batching]等技术以应对突然费用激增的组织来说，这可以使其更加经济。
+
+在我们仔细研究这个新的操作码本身之前，先来看看如何使用当前的比特币交易功能实现类似的事情。
+
+Alice 想支付给十个人，但当前的交易费用很高，所以她不想发送十笔单独的交易，甚至不想使用支付批处理来发送包含每个接收者输出的一笔交易。相反，她希望在不必为现在的十个输出支付上链费用的情况下，信任地承诺在未来支付给他们。因此，Alice 向每个接收者索要一个公钥，并创建一个未签名和未广播的*设置*交易，该交易使用 10-of-10 多重签名脚本支付这些公钥。然后她创建一个子交易，从多重签名输出支付给她最初想创建的 10 个输出。我们称这个子交易为*分配*交易。她要求所有接收者签署这个分配交易，并确保每个人都收到其他所有人的签名，然后她自己签署并广播设置交易。
+
+![一个设置交易支付预签名的分配交易](/img/posts/2019-05-tx-tree-1.png)
+
+当设置交易获得合理数量的确认时，Alice 就无法收回对 10 个接收者的付款了。只要每个接收者有一份分配交易及其他人的签名，就没有办法让任何接收者骗取其他接收者的付款。因此，即使实际上支付接收者的分配交易尚未广播或确认，这些付款也由已确认的设置交易保障。在任何时候，任何想花钱的接收者都可以广播分配交易并等待确认。
+
+这种技术允许付款人和接收人锁定一组付款在高费用期间，然后仅在费用较低时分配实际付款。根据撰写本文时的 Bitcoin Core 费用估算，任何有耐心等待一周交易确认的人（如上面的分配交易）都可以大幅节省费用。让我们在这个上下文中看看上面的例子。为了使后面的与 Taproot 的比较更公平，我们假设使用了一种形式的密钥和签名聚合，例如 [MuSig][] 或（理论上）多方 ECDSA（参见 [Newsletter #18][]）。
+
+{% assign p2wpkh = "A transaction spending one P2WPKH input to two P2WPKH outputs" %}
+{% assign batched11 = "A transaction spending one P2WPKH input to eleven P2WPKH outputs, or 9*(8+1+22) more bytes than the two-output P2WPKH transaction" %}
+{% assign batched10 = "A transaction spending one P2WPKH input to ten P2WPKH outputs, or 8*(8+1+22) more bytes than the two-output P2WPKH transaction" %}
+
+|                     | 独立付款              | 批量付款              | 现在承诺，以后分配        |
+|---------------------|-----------------------|-----------------------|-----------------------|
+| 即时（高费用）交易  | 10x<abbr title="{{p2wpkh}}">141 vbytes</abbr> | 1x<abbr title="{{batched11}}">420 vbytes</abbr> | 1x<abbr title="{{p2wpkh}}">141 vbytes</abbr> |
+| 费用（0.00142112 BTC/KvB） | 0.00204641 BTC         | 0.00059687 BTC         | 0.00020037 BTC         |
+| 延迟（低费用）交易  | ---                   | ---                   | 1x<abbr title="{{batched10}}">389 vbytes</abbr> |
+| 费用（0.00001014 BTC/KvB） | ---                   | ---                   | 0.00000394 BTC         |
+| 总 vbytes          | 1,410                 | 420                   | 530                   |
+| 总费用             | 0.00204641 BTC         | 0.00059687 BTC         | 0.00020431 BTC         |
+| **与前一列相比的节省** | ---                   | **71%**               | **66%**               |
+
+我们看到这种类型的信任延迟付款比支付批处理节省了 66%，比单独发送付款节省了 90%。请注意，在费用分层更大或接收者超过十个的情况下，节省可能会更大。
+
+### CheckOutputsHashVerify
```

Comment:
The en version seems a bit different from this.

Reply:
In our recent technical discussion, we found that "Trustless" should be translated as "无需信任" instead of "信任". There is a clear difference in meaning between the two. Initially, I did not recognize this point, but I have added links internally to correct the current content. I have noticed similar issues in subsequent content and will pay attention to them in future work.